fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/mixin/event/interaction/client/MinecraftClientMixin.java

```diff
@@ -157,4 +166,36 @@ private void injectUseEntityCallback(CallbackInfo ci, Hand[] hands, int i1, int
 			ci.cancel();
 		}
 	}
+
+	@Inject(
+			method = "handleInputEvents",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/client/network/ClientPlayerEntity;isUsingItem()Z",
+					ordinal = 0
+			)
+	)
+	private void injectHandleInputEventsForPreAttackCallback(CallbackInfo ci) {
+		if (options.attackKey.isPressed()) {
```

Comment:
Would it make sense to also fire the event if `timesPressed > 0`? Then listeners could check `if (key.isPressed())` if they needed to.

Reply:
If timesPressed > 0 then isPressed() must be true.

Reply:
Not necessarily if the click was very fast, right?

Reply:
In MC code, it always call `onKeyPressed` after `setKeyPressed` regardless how fast it clicks

Reply:
I still believe that's incorrect if the following code runs:
```java
// Click first
KeyBinding.setKeyPressed(key2, true);
KeyBinding.onKeyPressed(key2);
// Immediately release (same "input tick")
KeyBinding.setKeyPressed(key2, false);
```
In that case, we will have `timesPressed > 0`, but `isPressed()` will still be false.

Reply:
Multiple callbacks can execute in same tick. I see.

Reply:
Fixed